presto-verifier/src/test/java/com/facebook/presto/verifier/rewrite/TestQueryRewriter.java

```diff
@@ -571,6 +571,25 @@ public void testRewriteFunctionCalls()
                         "FROM test_table");
     }
 
+    @Test
+    public void testRewriteCasts()
+    {
+        VerifierConfig verifierConfig = new VerifierConfig().setFunctionSubstitutes(
+                "/cast(x as varchar)/cast(floor(x) as varchar)/");
```

Comment:
I know this is just testing the technical rewrite functionality, but this wouldn't necessarily be valid sql depending on the type. do you plan to use typeof() for the real substitution to make this conditional (e.g. `if(typeof(x) is DOUBLE OR typeof(x) is float, cast(floor(x) as varchar), cast(x as varchar)`)?

Reply:
Hi @rschlussel , yes, with real substitution in prod, we need the `typeof`, ex. `if(typeof(a) = 'double', cast(floor(cast(cast(a as varchar) as double)) as varchar), cast(a as varchar))`. Here I just put a very short pattern for this unit test.